### PR TITLE
Registration and login

### PR DESCRIPTION
**Pull request by nikolapantelic-ftn**
Added:
- Registration and login forms
- Form validation
- Knowledge node route guard
- Navbar trainee display

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -15,6 +15,15 @@
         &nbsp;&nbsp;Smart Tutor
       </span>
       <span fxFlex></span>
+      <span *ngIf="trainee">
+        <button mat-button [matMenuTriggerFor]="menu">{{trainee.studentIndex}}<mat-icon>expand_more</mat-icon></button>
+        <mat-menu #menu>
+          <button mat-menu-item (click)="onLogout()">Log out</button>
```
Comment (Luburic) [inline src/app/app.component.html:21]: Not something we need to address at this point since we are not sure it will become an issue, but still worth mentioning - the responsibility of the app component is to tie the main layout together.
Currently, another responsibility is emerging related to the toolbar, which is a sign that we should be thinking about creating a header component to encapsulate the login/logout coordination.

Still, I wouldn't perform any refactoring yet, but we should keep an eye on this and extract a component when the next piece of logic populates the app component.

**Review by Luburic — APPROVED** [review]
Looks good. We should wait for @milicasiriski to complete the backend PR and test out all the variations - empty name (front end validation can be bypassed by targeting endpoint directly or editing javascript), registering under an existing user and any additional edge cases we can think of.
